src/test/com/yetanalytics/datasim_test.clj

```diff
@@ -118,63 +120,143 @@
-      (is (every?
-           #(= bob-mailto (get-actor-mbox %))
-           ret))))
+    (let [input  (assoc-in const/simple-input [:parameters :max] 3)
+          result (generate-seq input
+                               ;; specify we only want the given agent(s)
+                               :select-agents [bob-mbox])]
+      (is (every? #(= bob-mailto (get-actor-mbox %))
+                  result))))
   (testing "Only actors in personae are generated"
-    (is (= #{alice-mailto bob-mailto fred-mailto}
-           (->> const/simple-input generate-seq (map get-actor-mbox) set))))
-  (testing "Can apply object override"
-    (let [ret (generate-seq (assoc const/simple-input
-                                   :alignments const/override-alignments)
-                            :select-agents [bob-mbox])]
-      (is (every? #(or (= override-activity-1 %)
-                       (= override-activity-2 %))
-                  (map get-object ret)))))
+    (let [result (generate-seq const/simple-input)]
+      (is (= #{alice-mailto bob-mailto fred-mailto}
+             (set (map get-actor-mbox result))))))
+  (testing "Respects pattern weights"
+    (let [alignments [{:id     "https://w3id.org/xapi/cmi5#waivedsession"
+                       :weight 1.0}
+                      {:id     "https://w3id.org/xapi/cmi5#noresultsession"
+                       :weight 0.0}
+                      {:id     "https://w3id.org/xapi/cmi5#failedsession"
+                       :weight 0.0}
+                      {:id     "https://w3id.org/xapi/cmi5#completionnosuccesssession"
+                       :weight 0.0}
+                      {:id     "https://w3id.org/xapi/cmi5#completionmaybefailedsession"
+                       :weight 0.0}
+                      {:id     "https://w3id.org/xapi/cmi5#passedsession"
+                       :weight 0.0}
+                      {:id     "https://w3id.org/xapi/cmi5#completionpassedsession"
+                       :weight 0.0}]
+          input      (update-in const/simple-input
+                                [:models 0 :alignments]
+                                into
+                                alignments)
+          result     (generate-seq input :select-agents [bob-mbox])
+          verbs      (map #(get-in % ["verb" "id"]) result)]
+      (is (every? #{"http://adlnet.gov/expapi/verbs/satisfied"
+                    "http://adlnet.gov/expapi/verbs/waived"}
+                  verbs))))
+  (testing "Respects activity weights"
+    (let [alignments [{:id "https://w3id.org/xapi/cmi5/activities/block"
+                       :weight 1.0}
+                      {:id "https://w3id.org/xapi/cmi5/activities/course"
+                       :weight 0.0}
+                      {:id "https://w3id.org/xapi/cmi5/activitytype/block"
+                       :weight 0.0}
+                      {:id "https://w3id.org/xapi/cmi5/activitytype/course"
+                       :weight 0.0}]
```

Comment:
Ok I'm going to call it: the fact that the user would have to account for any Activity Types _in the Statement Rules_ is going to be a pain. This is especially true since right now alignment weights _do not otherwise affect Statement Rules_ (i.e. selection from `any` or `all` do not take into account weights; this may be a feature we'd want to add).